### PR TITLE
Move measurevariable

### PR DESCRIPTION
**Pull request by Wenxing-hou**

### CODE REVIEW

File: SecurityPkg/Tcg/Tcg2Dxe/Tcg2Dxe.c [hunk 1]
```diff
@@ -2333,6 +2334,36 @@ MeasureAllSecureVariables (
     DEBUG ((DEBUG_INFO, "Skip measuring variable %s since it's deleted\n", EFI_IMAGE_SECURITY_DATABASE2));
   }
 
+  //
+  // Meaurement UEFI device signature database
```
Comment (jyao1) [inline SecurityPkg/Tcg/Tcg2Dxe/Tcg2Dxe.c:2338]: Please check AuthLib as well.

File: MdePkg/MdePkg.dec [hunk 2]
```diff
@@ -739,6 +739,12 @@
   ## Include/Protocol/SerilaIo.h
   gEfiSerialTerminalDeviceTypeGuid = { 0x6AD9A60F, 0x5815, 0x4C7C, { 0x8A, 0x10, 0x50, 0x53, 0xD2, 0xBF, 0x7A, 0x1B }}
 
+  # GUIDs defined in UEFI2.10
+  #
+  ## GUID used to specify section with devdb content
+  ## Include/Guid/DeviceAuthentication.h
+  gEdkiiDeviceSignatureDatabaseGuid  = {0xb9c2b4f4, 0xbf5f, 0x462d, {0x8a, 0xdf, 0xc5, 0xc7, 0xa, 0xc3, 0x5d, 0xad}}
```
Comment (jyao1) [inline MdePkg/MdePkg.dec]: why use Edkii as prefix?

File: MdeModulePkg/Universal/Variable/RuntimeDxe/Measurement.c [hunk 3]
```diff
@@ -32,6 +33,7 @@ VARIABLE_TYPE  mVariableType[] = {
   { EFI_IMAGE_SECURITY_DATABASE,  &gEfiImageSecurityDatabaseGuid },
   { EFI_IMAGE_SECURITY_DATABASE1, &gEfiImageSecurityDatabaseGuid },
   { EFI_IMAGE_SECURITY_DATABASE2, &gEfiImageSecurityDatabaseGuid },
+{ EDKII_DEVICE_SECURITY_DATABASE, &gEdkiiDeviceSignatureDatabaseGuid },
```
Comment (jyao1) [inline MdeModulePkg/Universal/Variable/RuntimeDxe/Measurement.c]: why use EDKII prefix?

File: SecurityPkg/Tcg/Tcg2Dxe/Tcg2Dxe.c (hunk 1 above)
Comment (Wenxing-hou) [reply SecurityPkg/Tcg/Tcg2Dxe/Tcg2Dxe.c:2338]: Thanks. I have added the measure.

File: MdePkg/MdePkg.dec (hunk 2 above)
Comment (Wenxing-hou) [reply MdePkg/MdePkg.dec]: I have fixed the name.

File: SecurityPkg/Tcg/Tcg2Dxe/Tcg2Dxe.c [hunk 4]
```diff
@@ -2333,6 +2334,36 @@ MeasureAllSecureVariables (
...
+  //
+  if ((PcdGet32 (PcdTcgPfpMeasurementRevision) >= TCG_EfiSpecIDEventStruct_SPEC_ERRATA_TPM2_REV_106) &&
+      (PcdGet8 (PcdEnableSpdmDeviceAuhenticaion) != 0)) {
+    Status = GetVariable2 (EDKII_DEVICE_SECURITY_DATABASE, &gEfiDeviceSignatureDatabaseGuid, &Data, &DataSize);
+    if (Status == EFI_SUCCESS) {
+      Status = MeasureVariable (
+                PCR_INDEX_FOR_SIGNATURE_DB,
+                EV_EFI_SPDM_DEVICE_POLICY,
+                EDKII_DEVICE_SECURITY_DATABASE,
```
Comment (Zhiqiang520) [inline SecurityPkg/Tcg/Tcg2Dxe/Tcg2Dxe.c]: I think here `EDKII_DEVICE_SECURITY_DATABASE` should be changed to `EFI_DEVICE_SECURITY_DATABASE`.

File: SecurityPkg/Tcg/Tcg2Dxe/Tcg2Dxe.c [hunk 5]
```diff
@@ -2333,6 +2334,36 @@ MeasureAllSecureVariables (
...
+                DataSize
+                );
+      FreePool (Data);
+    } else if (Status == EFI_NOT_FOUND) {
+      Data = NULL;
+      DataSize = 0;
+      Status = MeasureVariable (
+                PCR_INDEX_FOR_SIGNATURE_DB,
+                EV_EFI_SPDM_DEVICE_POLICY,
+                EDKII_DEVICE_SECURITY_DATABASE,
```
Comment (Zhiqiang520) [inline SecurityPkg/Tcg/Tcg2Dxe/Tcg2Dxe.c]: I think here `EDKII_DEVICE_SECURITY_DATABASE` should be changed to `EFI_DEVICE_SECURITY_DATABASE`.

File: SecurityPkg/Tcg/Tcg2Dxe/Tcg2Dxe.c (hunk 4 above)
Comment (Wenxing-hou) [reply SecurityPkg/Tcg/Tcg2Dxe/Tcg2Dxe.c]: Thanks. I have fixed the name error.

**Pull request merged**